src/main/java/modtrekt/ui/ProfileSidePanel.java

```diff
@@ -21,12 +24,27 @@ public class ProfileSidePanel extends UiPart<Region> {
     private Label inspirationalQuote;
 
     @FXML
-    private Label currentCapLabel;
+    private Label activeTasks;
 
-    @FXML
-    private Label semesterLabel;
-
-    public ProfileSidePanel() {
+    /**
+     * Creates a {@code ModuleListPanel} with the given {@code ObservableList}.
+     * @param logic
+     */
+    public ProfileSidePanel(Logic logic) {
         super(FXML);
+
+        refresh(logic);
+    }
+
+    /**
+     * Method to refresh user data.
+     */
+    public void refresh(Logic logic) {
+        // Count MC Completed
+        DoneModuleCommand.refresh(logic.getModuleList().getModuleList());
+        int totalCredits = DoneModuleCommand.getTotalCredits();
+        int totalTasks = logic.getTaskBook().getTaskList().filtered(Model.PREDICATE_HIDE_ARCHIVED_TASKS).size();
+        creditsCount.setText(totalCredits + " MCs");
```

Comment:
In the event that the sum is 1, it's better to display "1 MC" instead of "1 MCs" - check out StringUtil::pluralize".